kamon-scala-future/src/main/scala/kamon/instrumentation/futures/scala/ScalaFutureInstrumentation.scala

```diff
@@ -28,26 +32,99 @@ object ScalaFutureInstrumentation {
   val Component = "scala.future"
 
   /**
-    * Traces the execution of a block of code with a Delayed Span which takes into account the scheduling time when run
+    * Traces the execution of an asynchronous computation modeled by a Future. This function will create a Span when it
+    * starts executing and set that Span as current while the code that created the actual Future executes and then,
+    * automatically finish that Span when the created Future finishes execution.
+    *
+    * The purpose of this function is to make it easy to trace an asynchronous computation as a whole, even though such
+    * computation might be the result of applying several transformations to an initial Future. If you are interested in
+    * tracing the intermediate computations as well, take a look at the `traceBody` and `traceFun` functions bellow.
+    */
+  def trace[T](operationName: String, tags: TagSet = TagSet.Empty, metricTags: TagSet = TagSet.Empty)(future: => Future[T])
+      (implicit settings: Settings): Future[T] = {
+
+    val spanBuilder = Kamon.internalSpanBuilder(operationName, Component)
+    if(!settings.trackMetrics) spanBuilder.doNotTrackMetrics()
```

Comment:
I don't think this is necessary since you call the `trace(spanBuilder)` which will also call `doNotTrackMetrics` if it needs to.

Reply:
Correct, thanks!